lib/srv/server/azure_installer.go

```diff
@@ -81,6 +82,10 @@ func getInstallerScript(installerName, publicProxyAddr string) string {
 	nonce := make([]byte, 8)
 	// No big deal if rand.Read fails, the script is still valid.
 	_, _ = rand.Read(nonce)
-	return fmt.Sprintf("curl -s -L https://%s/v1/webapi/scripts/installer/%v | bash -s $@ #%x",
+	script := fmt.Sprintf("curl -s -L https://%s/v1/webapi/scripts/installer/%v | bash -s $@ #%x",
 		publicProxyAddr, installerName, nonce)
+	if clientID != "" {
+		script = fmt.Sprintf("export AZURE_CLIENT_ID=%q ; %s", clientID, script)
```

Comment:
Can we this passed to the installer web API as a parameter that would return the script with proper client ID baked in, instead of environment variable?